src/xarray_regrid/utils.py

```diff
@@ -238,6 +256,12 @@ def format_for_regrid(
         "lat": {"names": ["lat", "latitude"], "func": format_lat},
         "lon": {"names": ["lon", "longitude"], "func": format_lon},
     }
+
+    # Latitude padding adds a duplicate value which will undesirably
+    # alter statistical aggregations
+    if stats:
+        coord_handlers.pop("lat")
```

Comment:
Longitude padding should work fine here, but let me know if you can still reproduce the conversion to float.

Reply:
Weird. The problem seems to be gone now.